src/function/scalar/math/numeric.cpp

```diff
@@ -1161,4 +1162,98 @@ void EvenFun::RegisterFunction(BuiltinFunctions &set) {
 	                               ScalarFunction::UnaryFunction<double, double, EvenOperator>));
 }
 
+//===--------------------------------------------------------------------===//
+// gcd
+//===--------------------------------------------------------------------===//
+
+// should be replaced with std::gcd in a newer C++ standard
+template <class TA>
+TA Gcd(TA left, TA right) {
+	TA a = left;
+	TA b = right;
+
+	// This protects the following modulo operations from a corner case,
+	// where we would get a runtime error due to an integer overflow.
+	if ((left == NumericLimits<TA>::Minimum() && right == -1) ||
+	    (left == -1 && right == NumericLimits<TA>::Minimum())) {
+		return 1;
+	}
+
+	while (true) {
+		if (a == 0) {
+			return TryAbsOperator::Operation<TA, TA>(b);
+		}
+		b %= a;
+
+		if (b == 0) {
+			return TryAbsOperator::Operation<TA, TA>(a);
+		}
+		a %= b;
+	}
+}
+
+struct GCDOperator {
+	template <class TA, class TB, class TR>
+	static inline TR Operation(TA left, TB right) {
+		return Gcd(left, right);
+	}
+};
+
+void GCDFun::RegisterFunction(BuiltinFunctions &set) {
+	ScalarFunctionSet funcs("gcd");
+
+	funcs.AddFunction(ScalarFunction({LogicalType::TINYINT, LogicalType::TINYINT}, LogicalType::TINYINT,
+	                                 ScalarFunction::BinaryFunction<int8_t, int8_t, int8_t, GCDOperator>));
+	funcs.AddFunction(ScalarFunction({LogicalType::SMALLINT, LogicalType::SMALLINT}, LogicalType::SMALLINT,
+	                                 ScalarFunction::BinaryFunction<int16_t, int16_t, int16_t, GCDOperator>));
+	funcs.AddFunction(ScalarFunction({LogicalType::INTEGER, LogicalType::INTEGER}, LogicalType::INTEGER,
+	                                 ScalarFunction::BinaryFunction<int32_t, int32_t, int32_t, GCDOperator>));
+	funcs.AddFunction(ScalarFunction({LogicalType::BIGINT, LogicalType::BIGINT}, LogicalType::BIGINT,
```

Comment:
Type specialisations for smaller types  are only required for optimization purposes - as the function itself is likely to be quite slow adding another cast is likely not going to be a big cost. Could we remove the `TINYINT, SMALLINT and INTEGER` specializations?

Reply:
Sounds good to me. I removed the specializations for both GCD and LCM.